Circuit module enhancements

nhowe91 opened the pull request:
Hello,

This is a pretty big update to the circuits module, and implements most of logical circuit creation from the comments  on #1278 

Main major changes:

- Circuits have been renamed "Physical circuits", just to differentiate themselves from logical ones
- Physical circuit types have been moved to their own table, to allow for properties
- Physical circuits displays what logical circuits they are members of
- Logical circuit creation (editor is fairly basic at the moment, but it gets the job done)
- Logical circuit mapping on Google Maps, displaying the circuit type as well

Things this update does not include:

- Subcircuits (mentioned in #1278. DB is updated for it, just not implemented)
- Import / Export capabilities
- API support
- Migration script for database changes (not sure if circuits module is widely used)

nhowe1991 committed:
Modified circuit types. Moved to its own table to add ability to  pick colors and line options for placement on map

nhowe1991 committed:
Updated circuit pages to handle new types from the database table. Modified map generation to include circuit info and support mapping specific lists of circuits

nhowe1991 committed:
Bugfix for max size modal. jQuery appends the 'w' while opening the modal, so it couldn't find the popup class

nhowe1991 committed:
Groundwork for logical circuits. logic circuit editor frontend created

nhowe91 committed:
Logical circuit add/update/delete works now. Could use more validation

nhowe91 committed:
Start of logical circuit details.

nhowe91 committed:
Finished LC details. Fixed accidental edit.

nhowe91 committed:
Bugfix: logical circuit was still going to physical circuit page

nhowe91 committed:
Updating schema file

nhowe91 committed:
Added member_count column to logical circuits

nhowe91 committed:
Added html5 color picker

nhowe91 committed:
Fixed logical circuit map to grab the location for a device

nhowe91 committed:
Bugfix: LC Mappings weren't getting deleted

nhowe91 committed:
p. circuit report log. circuits they are a member of

nhowe91 committed:
Custom field support for logical circuits

nhowe91 committed:
LC custom field support

nhowe91 committed:
Duplicate circuit validation

nhowe91 committed:
comment cleanup

nhowe91 committed:
Merge pull request #1 from phpipam/master

Updating local master branch

nhowe91 committed:
Merge pull request #2 from phpipam/master

Update to 1.3.2 from master

nhowe91 committed:
Merge pull request #3 from nhowe91/master

Updating enhancement branch to 1.3.2

jamescat commented:
I haven't read through the changes, just your notes, but I love that you put the effort in to make the physical/logical circuits feature move forward.

However, I noticed right away that there are 555(!!) files changed in this commit / PR. I am assuming all the mode changes are actually unnecessary. It would probably be more agreeable to the maintainers (of whom I am not one) if you were to clean those out of your commit and update the PR, than the way it stands now.

nhowe91 committed:
Repairing file permissions

nhowe91 committed:
Merge branch 'circuit_module_enhancements' of https://github.com/nhowe91/phpipam into circuit_module_enhancements

nhowe91 commented:
Hey @jamescat, thanks for the help, still a little new to using git properly and didn't realize it kept track of mode changes (though I had wondered why it was showing so many files changed that I hadn't edited). 

Should be good to go now, only showing 23 files changed

GaryAllan commented:
Helo @jamescat,

Thanks for the PR. Could you strip out debugging?

    var_dump($all_locations);

jamescat commented:
Hi @GaryAllan, I'd be willing to, except for one small hiccup... I think you probably meant that debugging request for @nhowe91 -- I only made a comment to him about the number of files impacted, but haven't helped him with writing the changes themselves from this PR in any way.

@nhowe91 can you help him with the request he made above?

Also... I was going to make some other cleanup suggestions tomorrow after I could go through your code more carefully to see if I found any more... But since you'll probably be making this one for Gary, I thought these stylistic changes might be good to mention now too.

(Hopefully they don't sound out of line or critical. -- I'm not even sure why I'm paying more attention to this PR than any of the others that I don't usually read through. And I certainly haven't even written this much code for the project myself, so I hope you'll take them in the spirit in which they're given.)

app/admin/circuits/edit-circuit-submit.php:
Lines 43-44
These preceding 2 lines (before your change at line 45) are commented out by you (replaced by your code) in a very similar block within the file edit-circuit.php, and should probably be treated the same way here unless there is a code path still needing $type_desc -- but also see my comment below for that file about simply removing the lines.

Lines 118-126
Unless I'm overlooking some non-spacing change within these lines, the tab to space changes here would probably be better reverted, to stay in the style of the current developers.

app/admin/circuits/edit-circuit.php:
Lines 62-63
We all do the same thing often when coding (testing, reverting, retesting, etc.), but the commented out code here -- assuming the maintainers first agree that your code is a good way to proceed -- should probably just be removed in favor of your new code when committing, instead of leaving the old processes in the code as comments.

app/admin/circuits/edit-options-submit.php:
Lines 43-49, 80
Remove commented code -- see previous reasoning

I haven't reviewed your actual code path / logic flow changes in anything more than a very cursory manner myself, and hadn't yet even done that much for the couple of new files you added. But now that Gary has his eyes on this PR, I think I'm out, as he's a much more appropriate person to decide if anything needs adjusting or deserves commentary than I.

GaryAllan commented:
@nhowe91

To set timescales. I'm hoping to start reviewing/merging the open PRs in a few weeks (1-2months to complete).

jamescat commented on app/tools/circuits/all-circuits-map.php:
```diff
@@ -108,9 +148,30 @@
+                    $html[] = "  strokeWeight: 3";
+                    $html[] = "});";
+
+                    print implode("\n", $html);
```
I forgot to mention this in my recent comment on the PR... But I'm slightly concerned by this move of the "print implode" line to be inside the code block. -- It may be perfectly fine (I haven't fully reviewed the code paths myself), but it should be carefully checked for introducing any failed code paths when compared to the original line which was outside of the block.

nhowe91 committed:
Code cleanup

nhowe91 committed:
Cleanup spacing

not sure why git detected a difference here, doesn't look changed in local editor

nhowe91 committed:
Merge branch 'circuit_module_enhancements' of https://github.com/nhowe91/phpipam into circuit_module_enhancements

nhowe91 committed:
Cleanup of options

nhowe91 commented:
Hey @jamescat, its no problem on the suggestions I don't find them critical at all. I was trying to stay in the same style when doing these changes. It looks like though Atom is displaying my tabs as double spaces in my editor, so I tried to spot a few more place where things didn't look right and fix them.

@GaryAllan The debugging lines from the all-circuit-maps are now removed. I'm going to look over this again though because it was one of the first things I started changing,  I will also be double checking the schema file and my SQL script to update the DB structure.

nhowe91 committed:
Update SCHEMA.sql

nhowe91 committed:
Added check on physical circuit deletetion to make sure its not part of a logical circuit

nhowe91 committed:
Added def. to schema, renamed large_update and tested DB script

nhowe91 commented:
The DB update script is now working, It does the following:

- Creates circuit types, logical circuit, and logical mapping tables
- Retrieves in use circuit types and add them to new table with default values
- Alters circuits table to to support types and subcircuits for future update

I tested it on MariaDB 10.1.30 using 'mysql -u admin -p <dbname> db/alter_circuitchanges.sql', similar to the upgrade instructions on the PHP IPAM setup

nhowe91 committed:
PHP 5.4 bugfix with empty()

phpipam committed:
Update index.php

phpipam commented:
Hi, thanks for submitting PR and sorry for late review.

I will merge it to master in `Merge Nhowe91 circuit module enhancements to master` #2077 , there are some changes made:

- UI changes
- redone popup
- Rewritten some scripts
- Redone upgrade procedure completely because of procedures

Please note that for anyone using this table names have changed, I would suggest to drop all changes this PR to Database.

br
